### PR TITLE
[WFLY-6660]: Allow to dump Artemis journal to file using WildfFly Man…

### PR DESCRIPTION
**Pull request by ehsavoie**
…agement API

Jira: https://issues.redhat.com/browse/WFLY-6660

### CODE REVIEW

File: messaging/WFLY-6660_artemis_data_tools.adoc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+=== Other Interested Projects
+
+== Requirements
+
+The new operation 'data-print' will added  to the https://wildscribe.github.io/WildFly/21.0/subsystem/messaging-activemq/server/[/subsystem=messaging-activemq/server resource]. It will print a report about journal records and summary of existent records, as well a report on paging. This operation shouldn't be executed against a running broker.
+
+=== Hard Requirements
+
+Provide a new 'data-print' operation that will return a stream with the following parameters:
+ * safe a Boolean parameter with a default value of 'false' to print the data structure without showing the your data.
```
Comment (mnovak1) [inline messaging/WFLY-6660_artemis_data_tools.adoc:48]: I'm not sure what this parameter is meant to do. Could you provide more details. please?

File: messaging/WFLY-6660_artemis_data_tools.adoc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+=== Nice-to-Have Requirements
+
+ Add an `archive` Boolean parameter to return a zip of the journal dump instead of the plain text report to reduce network usage.
+
+=== Non-Requirements
+
+This operation will only work on the embedded borker and is not by any mean a way to get data from another Artemis installation (standalone or embeded) using the configuration of the broker it is run against.
+Support for JDBC journal is not a requirement as there is no JDBC connection available on admin-mode.
```
Comment (mnovak1) [inline messaging/WFLY-6660_artemis_data_tools.adoc:57]: I guess it means that there will be no way to print JDBC journal, right?

Comment (ehsavoie) [reply messaging/WFLY-6660_artemis_data_tools.adoc:57]: yes

File: messaging/WFLY-6660_artemis_data_tools.adoc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+----
+attachment save --file=/home/ehsavoie/tmp/test.zip --operation=/subsystem=messaging-activemq/server=default:print-data(archive=true)
+----
+
+== Test Plan
+
+== Community Documentation
+////
+Generally a feature should have documentation as part of the PR to wildfly master, or as a follow up PR if the feature is in wildfly-core. In some cases though the documentation belongs more in a component, or does not need any documentation. Indicate which of these will happen.
+////
```
Comment (bstansberry) [inline messaging/WFLY-6660_artemis_data_tools.adoc:79]: @ehsavoie This section needs completing. I think it needs coverage in the regular docs (not just management API description).

File: messaging/WFLY-6660_artemis_data_tools.adoc [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+=== Affected Projects or Components
+
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+The new operation 'data-print' will added  to the https://wildscribe.github.io/WildFly/21.0/subsystem/messaging-activemq/server/[/subsystem=messaging-activemq/server resource]. It will print a report about journal records and summary of existent records, as well a report on paging. This operation shouldn't be executed against a running broker.
```
Comment (jbliznak) [inline messaging/WFLY-6660_artemis_data_tools.adoc:43]: not existing link
> The WildFly Model References have moved to the WildFly documentation as of WildFly 19.

It seems the correct one would now be https://docs.wildfly.org/22/wildscribe/subsystem/messaging-activemq/server/

Comment (ehsavoie) [reply messaging/WFLY-6660_artemis_data_tools.adoc:43]: or remove the .0 in the version :)

Comment (jbliznak) [reply messaging/WFLY-6660_artemis_data_tools.adoc:43]: that won't help :)

404:
* https://wildscribe.github.io/WildFly/21.0/subsystem/messaging-activemq/server
* https://wildscribe.github.io/WildFly/21/subsystem/messaging-activemq/server

**Review by jbliznak — APPROVED** [review]

**Pull request merged**